enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,294 @@
+  If no policy exists in a namespace, use the current default of:
+
+    ```yaml
+    { "default": [{ "type": "insecureAcceptAnything" }] }
+    ```
+
+    If a single policy exists, use the default:
+    
+    ```yaml
+    { "default": [{"type": "reject"}] }
+    ```
+
+  - adds [/etc/containers/registries.d/*.yaml](https://github.com/containers/image/blob/main/docs/containers-registries.d.5.md) to allow matching sigstore signatures, for example:
+
+    ```yaml
+    docker:
+      my-registry/image:
+        use-sigstore-attachments: true
+    ```
+
+    Alternatively, we can enable it node wide:
+
+    ```yaml
+    default-docker:
+      use-sigstore-attachments: true
+    ```
+- merges cluster scoped with namespaced ImagePolicy as a json string and update `status` of each ImagePolicy.
+Once an ImagePolicy object gets created/updated, container runtime config controller will call the syncHandler.
+Once an ImagePolicy object gets deleted, container runtime config controller will delete the machine config and run the syncHandler
+
+#### Notes
+
+ImagePolicy is defined as namespaced CRD. `namespace: openshift-config` is used for cluster policy.
+
+### Risks and Mitigations
+
+Risk: The cluster scoped and namespace policies overwrite
+
+Mitigation: MCO container runtime config controller makes sure namespaced policies for an image cannot override global definitions when merging cluster openshift-config policies and namespaced policies into status field of ImagePolicy instances.   
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+MCO container runtime config controller can add unit tests and e2e tests.
+- unit test: verify the policies from ImagePolicy instances merged correctly and the controller writes correct format policy.json
+- e2e test: verify the MCO writes configuration to the correct location
+
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+In order to use ImagePolicy, an existing cluster will have to make on upgrade.
```

Comment:
Have to make what?  Can we expand this section?

From [the template][1], the following points seem worth considering:

> What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to keep previous behavior?

Are the CRI-O config knobs the new resource exposes something that some existing customers might  be managing via bare MachineConfig resources?  If so, how are conflicts resolved?  How are conflicts surfaced to a user who might be surprised by them?

[1]: https://github.com/openshift/enhancements/blob/4f18064f4ad9124efec465c59346e37171c0fcc9/guidelines/enhancement_template.md#upgrade--downgrade-strategy